exonum/src/api/mod.rs

```diff
@@ -106,37 +95,31 @@ impl From<io::Error> for ApiError {
     }
 }
 
-impl From<StorageError> for ApiError {
-    fn from(e: StorageError) -> ApiError {
+impl From<storage::Error> for ApiError {
+    fn from(e: storage::Error) -> ApiError {
         ApiError::Storage(e)
     }
 }
 
-impl From<FromHexError> for ApiError {
-    fn from(e: FromHexError) -> ApiError {
-        ApiError::FromHex(e)
-    }
-}
-
 impl From<ApiError> for IronError {
     fn from(e: ApiError) -> IronError {
-        use std::error::Error;
-
         let mut body = BTreeMap::new();
         body.insert("debug", format!("{:?}", e));
-        body.insert("description", e.description().to_string());
+        body.insert("description", e.to_string());
         let code = match e {
             ApiError::FileExists(hash) |
             ApiError::FileNotFound(hash) => {
                 body.insert("hash", encode_hex(&hash));
                 status::Conflict
             }
+            ApiError::BadRequest(..) => status::BadRequest,
+            ApiError::InternalError(..) => status::InternalServerError,
             _ => status::Conflict,
```

Comment:
`Unauthorized` and `NotFound` have obvious HTTP status codes as well. Also, I don't think that `Conflict` returned here matches its description in [RFC 7231](https://tools.ietf.org/html/rfc7231#page-60). I'd rather return `BadRequest` as a catch-all code.

Reply:
Yeah, the previous logic (which I more or less preserved) seems completely wrong. 

In fact, it is so wrong that I am suspicious that maybe this is deliberate? Could it be the case that we want to return the same status code for the purposes of information hiding?

Reply:
More reasonable status codes in https://github.com/exonum/exonum/pull/474/commits/bdfa835aad8da9205ccc8072fc71786af8655037